src/strategy.ts

```diff
@@ -57,38 +59,59 @@ export abstract class AbstractStrategy extends PassportStrategy {
       loggedOut: boolean;
     }) => {
       if (loggedOut) {
-        // Log out the current user no matter if we can verify the logged in user === logout requested user
-        req.logout();
-
-        // Check to see if we are logging out the user that is currently logged in to craft a proper IdP response
         if (profile != null) {
           if (this._saml == null) {
             throw new Error("Can't get logout response URL without a SAML provider defined.");
           }
 
-          const RelayState = req.query?.RelayState || req.body?.RelayState;
+          const getLogoutResponseUrl = this._saml.getLogoutResponseUrl;
+          // When logging out a user, use the consumer's `validate` function to check that
+          // the `profile` associated with the logout request resolves to the same user
+          // as the `profile` associated with the current session.
```

Comment:
Come to think of it this approach https://github.com/node-saml/passport-saml/blob/80fc3a2230874f8698c3e03a27818041a4d967fb/src/strategy.ts#L68-L70 would not work if verify function - like the one given at `passport-saml`'s [README.md](https://github.com/node-saml/passport-saml/blob/bba77bcc8030c633a2df5dd25f5c86f6caef5ce8/README.md) - uses some other information (assertion) from SAML login response than `nameID` to lookup user e.g. from DB.

`LogoutRequest` contains only `nameID` and `sessionIndex` thus `profile` constructed from `LogoutRequest` would not contain e.g. `email` information.

I.e. if `verify` function would be
```javascript
  function(profile, done) {
    findByEmail(profile.email, function(err, user) {
      if (err) {
        return done(err);
      }
      return done(null, user);
    });
  }
```
it would not resolve to same `user` object with the one that was resolved during login phase `verify` function processing.

Reply:
You are entirely correct. Since we don't currently have anyone running into this problem, I'm tempted to gloss over this until we can get some real-world use-cases. There are things that can be done with the IdP to help this matter, like including more data on SLO requests and duck-typing of `profile` can be used to branch this function internally for login/logout.

Reply:
I've implemented a second verify function to allow the user to keep the implementations separate. They will have to actually do the calculations to figure out which user is to be logged out.